analysis/datetime/flexible/flexible.go

```diff
@@ -44,6 +45,52 @@ func (p *DateTimeParser) ParseDateTime(input string) (time.Time, error) {
 	return time.Time{}, analysis.ErrInvalidDateTime
 }
 
+func validateLayout(layout string) bool {
+	validMagicNumbers := map[string]bool{
```

Comment:
map[string]struct{}{

Reply:
Also let's initialize this construct globally as opposed to setting up a map on every call to validateLayout.